### DISCUSSION TITLE
Error: Get config: Unable to establish a connection to query-engine-node-api library

### DISCUSSION THREAD

**Discussion by mirandaa1** [prisma/prisma#14128] [Q&A]
I got this error while trying to connect the project to SQLite Database using prisma and schema_prisma in remix app

**Comment by janpio** [comment]
Can you share the rest of the error message as well? There should be an actual reason given.

Please also share the version of Prisma you are using.

**Reply by mirandaa1** [reply]
that's all the error message though...
 prisma/client - "^3.14.0",
 prisma "^3.14.0"  and these are the versions of prisma

<img width="806" alt="image" src="https://user-images.githubusercontent.com/36954358/177269356-fd356be5-dbe3-4153-b8f0-f8c179344f1d.png">

in the image i was using prisma version 4, but after i switched to 3.14.0, still the same.

**Reply by mirandaa1** [reply]
@janpio if you could help me with this, i would appreciate it because I tried several manners to fix these missing files but still the same... 
thank you in advance.

**Reply by janpio** [reply]
The `details` line is the relevant one.

Can you please check if the file that is mentioned there is indeed missing and does not exist? Do other files exist there?

**Reply by mirandaa1** [reply]
@janpio the file query_engine-windows.dll.node exits in node_modules. You can check the other files there.
<img width="211" alt="image" src="https://user-images.githubusercontent.com/36954358/177753509-caea532f-e830-4eb2-9c44-b3ead41f06e3.png">

**Reply by janpio** [reply]
That is super weird - as the error message ten more or less is lying to you. Do you agree?

Do you ever have permission problems with this `OneDrive\Desktop` folder? Can you maybe try to install the project in your `Users\miran\Documents` folder instead? That one is definitely fully accessible all the time on Windows.

**Reply by mirandaa1** [reply]
@janpio yes of course I will try this solution.. and no I didn't have this kind of problem.

**Reply by mirandaa1** [reply]
@janpio unfortunately I got the same error...
<img width="695" alt="image" src="https://user-images.githubusercontent.com/36954358/177939915-745477bc-da41-4680-bc22-a45187393e8c.png">
I wouldn't insist about fixing this, but I need this for my thesis to be finished, that's why..

**Reply by janpio** [reply]
What is this Npm warning with "global" about? That should not be there.

Is your app on Github? Then we could try if we can reproduce the problem.

**Reply by mirandaa1** [reply]
yes the app is in the github, here you have the repo https://github.com/mirandaa1/remix-app.git

**Reply by janpio** [reply]
Hm, after cloning and running `yarn`, this works no problem for me:

````
C:\Users\Jan\Documents\throwaway\14128\remix-app>yarn prisma generate
yarn run v1.22.18
$ C:\Users\Jan\Documents\throwaway\14128\remix-app\node_modules\.bin\prisma generate
Prisma schema loaded from prisma\schema.prisma

✔ Generated Prisma Client (3.14.0 | library) to .\node_modules\@prisma\client in 95ms
You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client
```
import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()
```
Done in 1.80s.

C:\Users\Jan\Documents\throwaway\14128\remix-app>npx prisma generate
Prisma schema loaded from prisma\schema.prisma

✔ Generated Prisma Client (3.14.0 | library) to .\node_modules\@prisma\client in 41ms
You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client
```
import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()
```

````

Note how I can use both `yarn prisma` and `npx prisma` to generate a Client successfully.

This indicates the problem is somehow in your local configuration or setup on your computer, and not with Prisma or your project.

**Reply by mirandaa1** [reply]
@janpio hmmm really? I reseted my PC also and still the same. Since there is no problem with Prisma and that's good, what do you suggest me to follow ? p.s thanks a lot for your effort.

**Reply by janpio** [reply]
Have you tried using `yarn prisma` locally?

Optimally try another computer from a friend, or if that is not available a different user on Windows. Maybe reinstall yarn and Npm and make sure not anti virus or other tool could block access to that file.

**Reply by mirandaa1** [reply]
Yes I tried `yarn prisma` and still the same. I just tried from a friend, and yes it is working well from another Windows..
Okay, I will try now reinstalling yarn/npm. Thank you.

**Reply by janpio** [reply]
Sorry that we can help here more, but those specific installation problems are pretty much impossible to debug. _Something_ is making it think that it can not connect to that file, but there is not more information available.

**Reply by mirandaa1** [reply]
@janpio yes I understand, it's okay. I appreciate it.

**Reply by janpio** [reply]
What happens when you run `npx prisma -v`?

This here could actually be your problem and describe a possible solution: https://github.com/prisma/prisma/issues/9975#issuecomment-952944479 Can you please try that one?

**Reply by mirandaa1** [reply]
I tried it and I got this error..
<img width="960" alt="image" src="https://user-images.githubusercontent.com/36954358/178118788-e8c6b7e1-b1cd-4400-b9a8-75847fbbef43.png">

**Reply by mirandaa1** [reply]
@janpio yessss, it is fixed now after installing MSVCR. Thank you so so much :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
